test/solo/notifications-timeout.test.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+// oboe's notifier will send all log messages
+process.env.APPOPTICS_DEBUG_LEVEL = 3;
+
+const ao = require('../..');
+const notifications = ao.notifications;
+const expect = require('chai').expect;
+
+const messages = [];
+
+notifications.on('message', function (msg) {
+  messages.push(msg);
+})
+
+
+describe('notification function tests (long tests)', function () {
+  it('should receive at least 4 keep-alive messages in a minute', function (done) {
+    let keepAliveCount = 0;
+    this.timeout(70000);
+    setTimeout(function () {
+      messages.forEach(m => {
+        if (m.source === 'oboe' && m.type === 'keep-alive') {
+          keepAliveCount += 1;
+        }
+      });
+      expect(keepAliveCount).gte(4, 'at least 4 keep-alive messages');
+      done();
+    }, 60000);
+  });
+
+  it('should handle restarting if keep-alive messages do not arrive', function (done) {
```

Comment:
how do we know it is restarting?

Reply:
`(notifications.startCount > 1)`

Reply:
Am I blind?

Reply:
it's a lot of new stuff to absorb. the mind overloads after a while.